libtsuba/src/tsuba.cpp

```diff
@@ -270,24 +271,32 @@ tsuba::CreateSrcDestFromViewsForCopy(
 
 katana::Result<void>
 tsuba::CopyRDG(std::vector<std::pair<katana::Uri, katana::Uri>> src_dst_files) {
-  // TODO(vkarthik): add do_all loop
-  std::vector<uint64_t> manifest_uri_idxs;
-  for (uint64_t i = 0; i < src_dst_files.size(); i++) {
-    auto [src_file_uri, dst_file_uri] = src_dst_files[i];
-    // We save the names of all the manifest files and we write them out at the end.
-    if (tsuba::RDGManifest::IsManifestUri(src_file_uri)) {
-      manifest_uri_idxs.push_back(i);
-      continue;
-    }
-    tsuba::FileView fv;
-    KATANA_CHECKED(fv.Bind(src_file_uri.string(), true));
-    KATANA_CHECKED(
-        tsuba::FileStore(dst_file_uri.string(), fv.ptr<char>(), fv.size()));
-  }
+  katana::DynamicBitset manifest_uri_idxs;
+  katana::do_all(
+    katana::iterate(size_t{0}, src_dst_files.size()),
+        [&](size_t i) {
+      auto [src_file_uri, dst_file_uri] = src_dst_files[i];
+      // We save the names of all the manifest files and we write them out at the end.
+      if (tsuba::RDGManifest::IsManifestUri(src_file_uri)) {
+        manifest_uri_idxs.set(i);
+        return;
+      }
+      // NB: can't use KATANA_CHECKED here because of the do_all, so we keep track of the 
+      // error for now and then return it later
+      tsuba::FileView fv;
+      auto fv_res = fv.Bind(src_file_uri.string(), true);
+      if (!fv_res) {
+        KATANA_LOG_FATAL("Failed to bind {}", src_file_uri.string());
+      }
+      auto fs_res = tsuba::FileStore(dst_file_uri.string(), fv.ptr<char>(), fv.size());
+      if (!fs_res) {
+        KATANA_LOG_FATAL("Failed to store to {}", dst_file_uri.string());
+      }
```

Comment:
You don't seem to be collecting the errors to use later. You are just crashing. Please actually do what the comment says. We have a copyable version of ErrorInfo which should work for this.

Reply:
Oh right, so I should have updated the comment, but I was trying to store the error and return the error later but some of the collective calls are in `Network.h` which wasn't available in the open repo?